### PR TITLE
[Kindergaten Garden] Add explanation of Python specific details

### PR DESCRIPTION
**Pull request by astrieanna**
Adds track-specific details:
* Explain interface expected by tests.
* Link to Python docs for classes, constructors, and default parameters.

For Issue #2856

### CODE REVIEW

**Comment by astrieanna** [comment]
I looked at the housekeeping check that failed, but I don't understand why the process exited 1.

The section for Kindergarten Garden looks successful:

```
DEBUG:generator:kindergarten-garden: attempting render
DEBUG:generator:kindergarten-garden: writing render to tmp file /tmp/tmps8eiuheh
DEBUG:generator:kindergarten-garden: formatting tmp file /tmp/tmps8eiuheh
DEBUG:generator:kindergarten-garden: check passed
DEBUG:generator:kindergarten-garden: removing tmp file /tmp/tmps8eiuheh
```

File: exercises/practice/kindergarten-garden/.docs/instructions.append.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+would, in the tests, be represented as `Garden("VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV")`.
+
+To make this work, your class will need to implement a constructor (the `__init__` method).
+If you're not familiar with constructors, [class and instance objects][class vs instance objects in python] from the Python docs gives a more detailed explanation of the `__init__` method.
+
+
+## Default Parameters
+
+In some tests, a list of students is passed to the constructor. This should override the twelve student roster provided in the problem statement.
+This requires making both of these statements work:
```
Comment (BethanyG) [inline exercises/practice/kindergarten-garden/.docs/instructions.append.md:29]:
```suggestion
Both of these statements need to work with your `__init__() method:
```

File: exercises/practice/kindergarten-garden/.docs/instructions.append.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+```
+Garden("VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV") # Makes a garden based on the default 12-student roster
+Garden("VRCC\nVCGG", students=["Valorie", "Raven"]) # Makes a garden based on a 2-student roster
+```
+
+One approach is to make students a default parameter; the Python docs describe default parameters while explaining [function definitions][function definitions]
```
Comment (BethanyG) [inline exercises/practice/kindergarten-garden/.docs/instructions.append.md:36]:
```suggestion
One approach is to make the student `list` a [default argument][default argument values]; the Python docs describe `default parameters` in depth while explaining [function definitions][function definitions].
```

File: exercises/practice/kindergarten-garden/.docs/instructions.append.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+
+[classes in python]: https://docs.python.org/3/tutorial/classes.html
+[class vs instance objects in python]: https://docs.python.org/3/tutorial/classes.html#class-objects
+[function definitions]: https://docs.python.org/3/reference/compound_stmts.html#function-definitions
```
Comment (BethanyG) [inline exercises/practice/kindergarten-garden/.docs/instructions.append.md:41]:
```suggestion
[default argument values]: https://docs.python.org/3/tutorial/controlflow.html#default-argument-values
[function definitions]: https://docs.python.org/3/reference/compound_stmts.html#function-definitions
```

**Review by BethanyG — CHANGES_REQUESTED** [review]
Hi @astrieanna 👋🏽 

Thank you so much for submitting this! Not all of these suggestions are required - so please feel free to push back and discuss.  Overall, this looks really good.

**Comment by astrieanna** [comment]
Your suggestions are great; I especially liked how you smoothed out the constructors vs `__init__()` phrasing. I've accepted all the suggestions.

**Comment by BethanyG** [comment]
> I looked at the housekeeping check that failed, but I don't understand why the process exited 1.

I haven't looked at it yet.  I _suspect_ its related to some new test or edit introduced in `problem-specifications`.  Our CI is set up to sync to `problem-specs`, and then complain (fail) on any PR if we haven't updated something.  A not-so-subtle nag to keep things updated.  😅   

When you first submitted your PR, I was in the process of doing a big sync and update -- 110 files!   But since I hadn't merged that, the CI was nagging. 😄   Once I updated everything, I rebased your PR ... so now it' passing CI.

I'll approve and merge in just a few minutes....

File: exercises/practice/kindergarten-garden/.docs/instructions.append.md [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+
+To make this representation work, your `class` will need to implement an `__init__()` method.
+If you're not familiar with `__init__()` or _constructors_, [class and instance objects][class vs instance objects in python] from the Python docs gives a more detailed explanation.
+
+
+## Default Parameters
+
+In some tests, a `list` of students is passed as an argument to `__init__()`.
+This should override the twelve student roster provided in the problem statement.
+Both of these statements need to work with your `__init__() method:
```
Comment (BethanyG) [inline exercises/practice/kindergarten-garden/.docs/instructions.append.md:30]:
```suggestion
Both of these statements need to work with your `__init__()` method:
```

File: exercises/practice/kindergarten-garden/.docs/instructions.append.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+```Python
+# Make a garden based on the default 12-student roster.
+Garden("VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV") 
+
+# Make a garden based on a 2-student roster.
+Garden("VRCC\nVCGG", students=["Valorie", "Raven"])
```
Comment (BethanyG) [inline exercises/practice/kindergarten-garden/.docs/instructions.append.md:37]:
```suggestion
Garden("VRCC\nVCGG", students=["Valorie", "Raven"]) 
```
```

File: exercises/practice/kindergarten-garden/.docs/instructions.append.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+This should override the twelve student roster provided in the problem statement.
+Both of these statements need to work with your `__init__()` method:
+
+```Python
+# Make a garden based on the default 12-student roster.
+Garden("VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV") 
+
+# Make a garden based on a 2-student roster.
+Garden("VRCC\nVCGG", students=["Valorie", "Raven"]) 
+
```
Comment (BethanyG) [inline exercises/practice/kindergarten-garden/.docs/instructions.append.md:38]:
```suggestion
```Python
# Make a garden based on the default 12-student roster.
Garden("VRCGVVRVCGGCCGVRGCVCGCGV\nVRCCCGCRRGVCGCRVVCVGCGCV") 

# Make a garden based on a 2-student roster.
Garden("VRCC\nVCGG", students=["Valorie", "Raven"]) 
```

```

**Review by BethanyG — APPROVED** [review]
@astrieanna   Yay!  Thanks so much for submitting this.  I had to do one quick last push to close the code fence.  Otherwise, this looks great.  🌟 

Due to that last push, we'll have to wait for the final checks to run, and then I can merge.

**Pull request merged**